Datasnap/Classes/Utilities/Datasnap.m

```diff
@@ -77,45 +83,45 @@ - (id)initWithApiKey:(NSString*)apiKey
         apiKeySecret:(NSString*)apiKeySecret
       organizationId:(NSString*)organizationId
            projectId:(NSString*)projectId
-      eventQueueSize:(NSInteger)eventNum
+           idfaOptIn:(bool)optIn
+               email:(NSString*)email
  andVendorProperties:(VendorProperties*)vendorProperties
 {
     self.vendorProperties = vendorProperties;
     if (self = [self init]) {
         self.organizationId = organizationId;
         self.projectId = projectId;
-        self.eventQueue = [[EventQueue alloc] initWithSize:eventNum];
         self.api = [[DatasnapAPI alloc] initWithKey:apiKey secret:apiKeySecret];
-        self.baseClient = [[BaseClient alloc] init];
+        self.baseClient = [BaseClient new];
         self.baseClient.projectId = projectId;
         self.baseClient.organizationId = organizationId;
+        self.optIn = optIn;
+        self.email = email;
     }
     [self initializeData];
     return self;
 }
-
 + (id)sharedClient
 {
     static Datasnap* sharedClient = nil;
     static dispatch_once_t onceToken;
     dispatch_once(&onceToken, ^{
-        sharedClient = [[self alloc] init];
+        sharedClient = [self new];
     });
     return sharedClient;
 }
-
 - (void)initializeData
 {
-    self.device = [[Device alloc] init];
+    self.device = [Device new];
     [self setDeviceAndReturn:self.device];
-    self.deviceInfo = [[DeviceInfo alloc] init];
+    self.deviceInfo = [DeviceInfo new];
     self.deviceInfo.device = self.device;
     self.baseClient.deviceInfo = self.deviceInfo;
-    self.user = [[User alloc] init];
+    self.user = [User new];
     [self.user initializeUser:self.user];
     self.baseClient.user = self.user;
-    self.identifier = [[Identifier alloc] init];
-    if ([[NSBundle mainBundle] objectForInfoDictionaryKey:@"SEND_ADVERTISER_ID"]) {
+    self.identifier = [Identifier new];
```

Comment:
Creating an Identifier with new and then setting properties seems like it puts a lot of unnecessary code in here. This should also be some sort of custom initializer.